component/src/main/java/org/wso2/extension/siddhi/gpl/execution/streamingml/classification/hoeffdingtree/util/AdaptiveHoeffdingTreeModel.java

```diff
@@ -182,14 +183,19 @@ private Instance createMOAInstance(double[] cepEvent) {
     }
 
     private InstancesHeader createMOAInstanceHeader(int numberOfAttributes) {
-        FastVector headerAttributes = new FastVector();
+        ArrayList<Attribute> attributes = new ArrayList<Attribute>();
         for (int i = 0; i < numberOfAttributes - 1; i++) {
-            headerAttributes.addElement(
-                    new Attribute("att_" + i));
+            attributes.add(new Attribute("numeric" + (i + 1)));
+        }
+        // Add class value
+        ArrayList<String> classLabels = new ArrayList<String>();
+        for (int i = 0; i < this.noOfClasses; i++) {
+            classLabels.add("class" + (i + 1));
         }
-        InstancesHeader streamHeader = new InstancesHeader(new Instances
-                (this.getCLICreationString(InstanceStream.class), headerAttributes, 0));
-        streamHeader.setClassIndex(streamHeader.numAttributes());
+        attributes.add(new Attribute("class", classLabels));
+        InstancesHeader streamHeader = new InstancesHeader(new Instances(getCLICreationString(InstanceStream.class),
+                attributes, 0));
+        streamHeader.setClassIndex(noOfFeatures - 1);
```

Comment:
Can you explain the logic here?

Reply:
It's creating the stream header(stream definition) needed for the model. Used moa Attributes instead using a FastVector

Reply:
I want to no about the noOfFeatures - 1 logic.

Reply:
Setting the index of the taget value for the training case